source/gravity_model/radial.cc

```diff
@@ -120,8 +120,8 @@ namespace aspect
         return Tensor<1,dim>();
 
       const double depth = this->get_geometry_model().depth(p);
-      return  (-magnitude_at_surface * p/p.norm() *
-               (1.0 - depth/this->get_geometry_model().maximal_depth()));
+      return  (-magnitude_at_surface  * p/p.norm() * (1.0 - depth/this->get_geometry_model().maximal_depth()))
+              + (-magnitude_at_bottom * p/p.norm() * (depth/this->get_geometry_model().maximal_depth()));
```

Comment:
You can pull the factor `p/p.norm()` out of the sum. It is really just the unit outward vector whereas the rest of the formula determines the magnitude.